xray/cli.go

```diff
@@ -114,3 +145,81 @@ func newXrCurlCommand(c *cli.Context) (*curl.XrCurlCommand, error) {
 	xrCurlCommand.SetUrl(xrDetails.XrayUrl)
 	return xrCurlCommand, err
 }
+
+func auditNpmCmd(c *cli.Context) error {
+	serverDetailes, err := CreateServerDetailsWithConfigOffer(c)
+	if err != nil {
+		return err
+	}
+	var typeRestriction = npmutils.All
+	switch c.String("type-restriction") {
+	case "devOnly":
+		typeRestriction = npmutils.DevOnly
+	case "prodOnly":
+		typeRestriction = npmutils.ProdOnly
+	}
+	auditNpmCmd := audit.NewAuditNpmCommand().SetServerDetails(serverDetailes).SetNpmTypeRestriction(typeRestriction).SetTargetRepoPath(c.String("repo-path"))
+	// Handle Xray's contex flags
+	if c.String("watches") != "" {
+		auditNpmCmd.SetWatches(strings.Split(c.String("watches"), ","))
+	} else if c.String("project") != "" {
```

Comment:
Since it looks like either "project" or "watches" is picked up, but not both, let's add a validation to make sure two er not sent by the user.